Extension/src/Debugger/configurationProvider.ts

```diff
@@ -545,6 +547,25 @@ export class DebugConfigurationProvider implements vscode.DebugConfigurationProv
             configs = configs.concat(await this.provideDebugConfigurationsForType(DebuggerType.cppvsdbg, folder));
         }
 
+        if (folder) {
+            // Get existing debug configurations from launch.json.
+            const existingConfigs: vscode.DebugConfiguration[] | undefined = (await this.getLaunchConfigs(folder))?.map(config => ({
+                name: config.name,
+                type: config.type,
+                request: config.request,
+                detail: config.detail ? config.detail : localize("pre.Launch.Task", "preLaunchTask: {0}", config.preLaunchTask),
+                preLaunchTask: config.preLaunchTask,
+                existing: TaskConfigStatus.configured
+            }));
+
+            const areEqual = (config1: vscode.DebugConfiguration, config2: vscode.DebugConfiguration): boolean =>
+                (config1.name === config2.name && config1.preLaunchTask === config2.preLaunchTask
+                && config1.type === config2.type && config1.request === config2.request);
+            // Remove the detected configs that are already configured once in launch.json.
+            const dedupExistingConfigs: vscode.DebugConfiguration[] = configs.filter(detectedConfig => existingConfigs ? !existingConfigs.some(config => areEqual(config, detectedConfig)) : true);
```

Comment:
you should be able to simplify this to not need the `? : true` by just returning `!existingConfigs?.some(...)` instead. (`?.` replaces the `.`)

Reply:
this `existingConfigs `could be undefined, I added an if statement and then simplified this.

Reply:
> this `existingConfigs `could be undefined, I added an if statement and then simplified this.

Yes, that is what the `?.` does.  If the variable is undefined, it will return undefined instead of trying to call the `some` function.  When you add the `!` in front of it, it turns into `true`, so that case is handled.

Reply:
What you've done with the if statement also works.